src/components/ProfileHeader.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from "react";
+import "../styles/ProfileHeader.scss";
+import EditOutlinedIcon from "@material-ui/icons/EditOutlined";
+import Dropdown from "react-bootstrap/Dropdown";
+import DropdownButton from "react-bootstrap/DropdownButton";
+
+class ProfileHeader extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: "Username",
+      remote: false,
+      friend: true,
+      following: true,
+      self: false,
+      // self could be checked using the userID and the loggedIn user's id
+      // to do: fetch the user
+    };
+  }
+
+  handleEditButtonClick = () => {
+    // eslint-disable-next-line no-alert
+    alert("todo");
+  };
+
+  renderStatus = () => {
+    const { self, following, friend } = this.state;
+    if (!self) {
+      if (!friend && !following) {
+        return (this.renderFollowButton());
+      }
+      if (!friend && following) {
+        return (this.renderDropDown(false));
+      }
+      return (this.renderDropDown(true));
+    }
+    return (this.renderEditProfileButton());
+  };
+
+  renderEditProfileButton = () => (
+    <button
+      type="button"
+      className="edit-profile-button"
+      onClick={this.handleEditButtonClick}
+    >
+      <EditOutlinedIcon className="edit-icon" />
+      <span>EDIT PROFILE</span>
+    </button>
+  );
+
+  handleFollow = () => {
+    this.setState({ following: true });
+  };
+
+  renderDropDown = (isFriend) => (
+    <DropdownButton
+      id="friend-status"
+      title={isFriend === true ? "FRIENDS" : "FOLLOWING"}
+      drop="down"
+      alignRight
+    >
+      <Dropdown.Item onClick={this.handleUnStatus}>{isFriend === true ? "Unfriend" : "Unfollow"}</Dropdown.Item>
+    </DropdownButton>
+  );
+
+  handleUnStatus = () => {
+    const { friend } = this.state;
+    if (friend) {
+      this.setState({ friend: false });
+    }
+    this.setState({ following: false });
```

Comment:
we should combine the setState calls into one since currently if friend is true then it will call setState 2 times causing 2 re-renders.